src/components/StatesButton/StatesButton.tsx

```diff
@@ -44,18 +62,56 @@ export class StatesButton extends React.Component<
     );
   }
 
+  private renderContent(): React.ReactElement | string {
+    const {
+      state,
+      idleContent,
+      inProgressContent,
+      failureContent,
+      successContent,
+    } = this.props;
+
+    switch (state) {
+      case BUTTON_STATES.IDLE:
+        return idleContent;
+      case BUTTON_STATES.IN_PROGRESS:
+        return inProgressContent;
+      case BUTTON_STATES.FAILURE:
+        return failureContent;
+      case BUTTON_STATES.SUCCESS:
+        return successContent ? successContent : this.renderCheck();
+      default:
+        return idleContent;
+    }
+  }
+
   public render() {
-    const { text, disabled, dataHook, ...rest } = this.props;
-    const { success } = this.state;
+    const {
+      state,
+      disabled,
+      dataHook,
+      onClick,
+      idleContent,
+      inProgressContent,
+      failureContent,
+      successContent,
+      onSuccessEnd,
+      ...rest
+    } = this.props;
+    const inProgress = state === BUTTON_STATES.IN_PROGRESS;
+
     return (
       <Button
         data-hook={dataHook}
-        disabled={disabled}
-        {...rest}
+        disabled={this.isDisabled()}
```

Comment:
So from what I talked with @NeilWix, we don't want the button to be disabled.
The `aria-busy` is enough.
We do, however, need to block the `onClick` from being called.

Reply:
According to Johanna's spec, it is supposed to be disabled and also look disabled when in progress.
Basically we can leave it to the consumer to decide, but I'm not sure if that's what Johanna wants..
WDYT?

Reply:
Johanna's spec is design only.
This is the a11y behaviour that we agreed with Neil.
BTW, she told she's not sure she wants the button to look disabled.
And I told her that technically the button won't be disabled.